forgerock-core/src/main/java/org/forgerock/android/auth/NetworkConfig.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth;
+
+import android.content.Context;
+
+import org.forgerock.android.core.R;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+import okhttp3.CookieJar;
+import okhttp3.Interceptor;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Manages Network configuration information
+ */
+@EqualsAndHashCode
+@Getter
+class NetworkConfig {
+
+    private String url;
```

Comment:
Is the url the combination of host, port and realm? I see 'host' as a Network configuration attribute, not sure about 'url' and 'realm'. Do you think makes sense move url and realm to ServerConfig?

Is possible to make the URL nullable? In the Authenticator SDK, for example, we don't need the URL to build a new OkHttpClient. In order to use it today, would be necessary to use the OkHttpClientProvider.getInstance().lookup() every time a new Push authentication mechanism is added, once the registration URL may be different for each mechanism.

Reply:
Make sense, will make the changes.